streaming_providers/routes.py

```diff
@@ -136,6 +139,83 @@ async def streaming_provider_endpoint(
     )
 
 
+# This endpoint is to proxy the debrid provider via this instance.
+@router.head("/{secret_str}/proxy_stream", tags=["streaming_provider"])
+@router.get("/{secret_str}/proxy_stream", tags=["streaming_provider"])
+@wrappers.exclude_rate_limit
+@wrappers.auth_required
+async def proxy_streaming_provider_endpoint(
+    secret_str: str,
+    info_hash: str,
+    response: Response,
+    request: Request,
+    season: int = None,
+    episode: int = None,
+):
+    response.headers.update(const.NO_CACHE_HEADERS)
+
+    user_data = request.scope.get("user", crypto.decrypt_user_data(secret_str))
+    if not user_data.streaming_provider:
```

Comment:
+ check the `user_data.proxy_debrid_stream is True`

Reply:
It's being checked below, I had to get user data to proceed, but changed the code a bit, the validations to user data is being done already, so moving this a bit below after calling `streaming_provider_endpoint` so that we can remove this check.